README.md
<p align="center"><img src="screenshots/reviewer.gif" height=500></p>

<h2 align="center">Puppy Reinforcement for Anki</h2>

<p align="center">
<a title="Latest (pre-)release" href="https://github.com/glutanimate/puppy-reinforcement/releases"><img src ="https://img.shields.io/github/release-pre/glutanimate/puppy-reinforcement.svg?colorB=brightgreen"></a>
<a title="License: GNU AGPLv3" href="https://github.com/glutanimate/puppy-reinforcement/blob/master/LICENSE"><img  src="https://img.shields.io/badge/license-GNU AGPLv3-green.svg"></a>
<a title="Rate on AnkiWeb" href="https://ankiweb.net/shared/info/1722658993"><img src="https://glutanimate.com/logos/ankiweb-rate.svg"></a>
<br>
<a title="Buy me a coffee :)" href="https://ko-fi.com/X8X0L4YV"><img src="https://img.shields.io/badge/ko--fi-contribute-%23579ebd.svg"></a>
<a title="Support me on Patreon :D" href="https://www.patreon.com/bePatron?u=7522179"><img src="https://img.shields.io/badge/patreon-support-%23f96854.svg"></a>
<a title="Follow me on Twitter" href="https://twitter.com/intent/user?screen_name=glutanimate"><img src="https://img.shields.io/twitter/follow/glutanimate.svg"></a>
</p>

> Everything's better with puppies!

An add-on for the spaced-repetition flashcard app [Anki](https://apps.ankiweb.net/) that encourages learners with pictures of cute puppies, following the principles of intermittent positive reinforcement.

### Table of Contents <!-- omit in toc -->

<!-- MarkdownTOC levels="1,2,3" -->

- [Video](#video)
- [Installation](#installation)
- [Documentation](#documentation)
- [Building](#building)
- [Contributing](#contributing)
- [License and Credits](#license-and-credits)

<!-- /MarkdownTOC -->

###  Video

A quick demo of the add-on (alongside other similar add-ons related to gamification):

[![YouTube: Anki add-on guide: Gamify Your Reviews](https://i.ytimg.com/vi/UkveLkAgXiM/mqdefault.jpg)](https://youtu.be/UkveLkAgXiM)

### Installation

#### AnkiWeb <!-- omit in toc -->

The easiest way to install Puppy Reinforcement is through [AnkiWeb](https://ankiweb.net/shared/info/1722658993).

#### Manual installation <!-- omit in toc -->

1. Download the latest `.ankiaddon` file from the [releases tab](https://github.com/glutanimate/puppy-reinforcement/releases) (you might need to click on *Assets* below the description to reveal the download links)
2. Open the folder where your downloads are located and double-click on the downloaded `.ankiaddon` file.
3. Follow the installation prompt and restart Anki if it asks you to

<details>

<summary><i>Alternate option</i></summary>

1. Download the latest `.ankiaddon` package from the [releases tab](https://github.com/glutanimate/puppy-reinforcement/releases) (you might need to click on *Assets* below the description to reveal the download links)
2. From Anki's main window, head to *Tools* → *Add-ons*
3. Drag-and-drop the `.ankiaddon` package onto the add-ons list
4. Restart Anki

Video summary:

<img src="https://raw.githubusercontent.com/glutanimate/docs/master/anki/add-ons/media/ankiaddon-installation-macos.gif" width=640>

</details>

### Documentation

For further information on the use of this add-on please check out [the description text](docs/description.md) for AnkiWeb.

### Building

With [Anki add-on builder](https://github.com/glutanimate/anki-addon-builder/) installed:

    git clone https://github.com/glutanimate/puppy-reinforcement.git
    cd puppy-reinforcement
    aab build

For more information on the build process please refer to [`aab`'s documentation](https://github.com/glutanimate/anki-addon-builder/#usage).

### Contributing

Contributions are welcome! Please review the [contribution guidelines](./CONTRIBUTING.md) on how to:

- Report issues
- File pull requests
- Support the project as a non-developer

### License and Credits

*Puppy Reinforcement* is *Copyright © 2016-2023 [Aristotelis P.](https://glutanimate.com/) (Glutanimate)*

With code contributions from: [@zjosua](https://github.com/zjosua) (thanks!)

Puppy Reinforcement is free and open-source software. The add-on code that runs within Anki is released under the GNU AGPLv3 license, extended by a number of additional terms. For more information please see the [LICENSE](https://github.com/glutanimate/puppy-reinforcement/blob/master/LICENSE) file that accompanied this program.

Please note that this program uses the [Libaddon](https://github.com/glutanimate/anki-libaddon/) library which comes with [its own additional terms extending the GNU AGPLv3 license](https://github.com/glutanimate/puppy-reinforcement/blob/master/src/puppy_reinforcement/libaddon/LICENSE). You may only copy, distribute, or modify the present compilation of this program with Libaddon under the combined licensing terms specified by both licenses.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.

----

<b>
<div align="center">The continued development of this add-on is made possible <br>thanks to my <a href="https://www.patreon.com/glutanimate">Patreon</a>  supporters. You guys rock ❤️ !</div>
</b>


CHANGELOG.md
# Changelog

All notable changes to Puppy Reinforcement will be documented here. You can click on each release number to be directed to a detailed log of all code commits for that particular release. The download links will direct you to the GitHub release page, allowing you to manually install a release if you want.

If you enjoy Puppy Reinforcement, please consider supporting my work on Patreon, or by buying me a cup of coffee :coffee::

<p align="center">
<a href="https://www.patreon.com/glutanimate" rel="nofollow" title="Support me on Patreon 😄"><img src="https://glutanimate.com/logos/patreon_button.svg"></a>      <a href="https://ko-fi.com/X8X0L4YV" rel="nofollow" title="Buy me a coffee 😊"><img src="https://glutanimate.com/logos/kofi_button.svg"></a>
</p>

:heart: My heartfelt thanks goes out to everyone who has supported this add-on through their tips, contributions, or any other means (you know who you are!). All of this would not have been possible without you. Thank you for being awesome!

## [Unreleased]

## [1.1.1] - 2023-11-16

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v1.1.1)

### Fixed

- Fixed a compatibility issue with Anki 2.1.49

## [1.1.0] - 2023-11-14

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v1.1.0)

### Added

- Added support for Anki 23.10+
- The tooltip position is now fully customizable, both horizontally and vertically (thanks to eotini23 on Patreon for the idea!)
- Add option to disable tooltips while reviewing cards (thanks to Jamie on Patreon for the idea!)
- Add option to reset the counter once a new day starts (thanks to p4nix on Patreon for the idea!)

### Fixed

- Suppress warning on unclosed windows

### Changed

- Rewrote notification/tooltip module from scratch, paving the way for the new features listed above, and many more to come!

## [1.0.0] - 2020-03-02

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v1.0.0)

It took them a while to waddle up the version steps with their tiny legs, but your friendly neighborhood puppies are finally here! Here is everything that's new in **Puppy Reinforcement v1.0.0**:

### Added

![](https://raw.githubusercontent.com/glutanimate/puppy-reinforcement/master/screenshots/addcards.gif)

- Anki 2.1 compatibility (thanks to @zjosua for his help with this!, #77)
- Optional support for showing puppies while creating new cards. Controlled by the `count_adding` option in the settings (thanks to @zjosua for the implementation and /u/xTheKezio / @ijgnd for the idea, #12)
- Support for more image file types and case-insensitive file extension detection
- A more advanced randomization approach that spreads the chances of particular images being drawn up evenly (#8)

### Fixed

- Added workaround for add-on import bug present in Anki versions <= 2.1.14
- Print "card" instead of "cards" if count singular

### Changed

- Rewrote large parts of the add-on, making it a lot more maintainable for the future
- Dropped Anki 2.0 support (legacy versions remain downloadable on AnkiWeb / GitHub)
- Added support for the new add-on hooks system that will be introduced with the upcoming Anki 2.1.20 release. This should make the add-on fairly future-proof.

## [1.0.0-beta.2] - 2020-02-05

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v1.0.0-beta.2)

### Fixed

- Fixed compatibility with Visual Feedback and other add-ons using newer versions of libaddon
- Print "card" instead of "cards" if count singular

## [1.0.0-beta.1] - 2020-01-28

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v1.0.0-beta.1)

### Added

- Anki 2.1 compatibility (thanks to @zjosua for his help with this!, #77)
- Optional support for showing puppies while creating new cards. Controlled by the `count_adding` option in the settings (thanks to @zjosua for the implementation and /u/xTheKezio / @ijgnd for the idea, #12)
- Support for more image file types and case-insensitive file extension detection
- A more advanced randomization approach that spreads the chances of particular images being drawn up evenly (#8)

### Fixed

- Added workaround for add-on import bug present in Anki versions <= 2.1.14

### Changed

- Rewrote large parts of the add-on, making it a lot more maintainable for the future
- Dropped Anki 2.0 support (legacy versions remain downloadable on AnkiWeb / GitHub)
- Added support for the new add-on hooks system that will be introduced with the upcoming Anki 2.1.20 release. This should make the add-on fairly future-proof.

## [0.2.0-alpha.1] - 2019-06-09

### [Download](https://github.com/glutanimate/puppy-reinforcement/releases/tag/v0.2.0-alpha.1)

**Important note**: As this release completely overhauls the add-on structure, you will have to uninstall any existing versions of the add-on before updating. Otherwise you might end up with duplicate versions of the add-on that would interfere with each other. *If you have customized the add-on with your own pictures or other adjustments, please make sure to back them up before deleting the old version!*

### Added

- Anki 2.1 compatibility (thanks to [@zjosua](https://github.com/zjosua) for his help with this!, #77)

### Changed

- Refactored add-on to improve stability and maintainability

## 0.1.2 - 2017-08-08 

### Added

- New options: duration, image size, card limits (thanks to Lucas on YT for the ideas!)

## 0.1.1 - 2017-08-06

### Changed

- Update license

## 0.1.0 - 2016-11-18

### Added

- Initial release of Puppy Reinforcement

[Unreleased]: https://github.com/glutanimate/puppy-reinforcement/compare/v1.0.0...HEAD
[1.1.1]: https://github.com/glutanimate/puppy-reinforcement/compare/v1.1.0...v1.1.1
[1.1.0]: https://github.com/glutanimate/puppy-reinforcement/compare/v1.0.0...v1.1.0
[1.0.0]: https://github.com/glutanimate/puppy-reinforcement/compare/v1.0.0-beta.2...v1.0.0

src/puppy_reinforcement/consts.py
# -*- coding: utf-8 -*-

# Puppy Reinforcement Add-on for Anki
#
# Copyright (C) 2016-2023  Aristotelis P. <https://glutanimate.com/>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version, with the additions
# listed at the end of the license file that accompanied this program.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#
# NOTE: This program is subject to certain additional terms pursuant to
# Section 7 of the GNU Affero General Public License.  You should have
# received a copy of these additional terms immediately following the
# terms and conditions of the GNU Affero General Public License that
# accompanied this program.
#
# If not, please request a copy through one of the means of contact
# listed here: <https://glutanimate.com/contact/>.
#
# Any modifications to this file must keep this entire header intact.

"""
Addon-wide constants
"""

from ._version import __version__

__all__ = ["ADDON"]

# PROPERTIES DESCRIBING ADDON

class ADDON:
    """Class storing general add-on properties
    Property names need to be all-uppercase with no leading underscores
    """

    NAME = "Puppy Reinforcement"
    MODULE = "puppy_reinforcement"
    ID = "1722658993"
    VERSION = __version__
    LICENSE = "GNU AGPLv3"
    AUTHORS = (
        {
            "name": "Aristotelis P. (Glutanimate)",
            "years": "2016-2023",
            "contact": "https://glutanimate.com",
        },
    )
    AUTHOR_MAIL = "ankiglutanimate@gmail.com"
    LIBRARIES = ()
    CONTRIBUTORS = ("zjosua",)
    SPONSORS = ()
    LINKS = {
        "patreon": "https://www.patreon.com/glutanimate",
        "bepatron": "https://www.patreon.com/bePatron?u=7522179",
        "coffee": "http://ko-fi.com/glutanimate",
        "description": "https://ankiweb.net/shared/info/{}".format(ID),
        "rate": "https://ankiweb.net/shared/review/{}".format(ID),
        "twitter": "https://twitter.com/glutanimate",
        "youtube": "https://www.youtube.com/c/glutanimate",
        "help": "",
    }
